topics/rsa-keys-generation.md

```diff
@@ -1,91 +1,108 @@
+<show-structure for="chapter" depth="2"/>
 
-RS256, part of the RSA (Rivest–Shamir–Adleman) encryption algorithm, utilizes SHA-256 for hashing and a key (usually 2048-bit, 4096-bit or higher) to secure digital communications. 
+<var name="example_name" value="auth-jwt-rs256"/>
 
-In the realm of [JSON Web Token](https://jwt.io/) authentication, RS256 plays a crucial role since the integrity and authenticity of JWTs can be verified through signature mechanisms, such as RS256, where a public/private key pair is employed. This ensures that the information contained within the token remains tamper-proof and trustworthy.
+<tldr>
+<include from="lib.topic" element-id="download_example"/>
+</tldr>
 
-The following sections will cover how such keys are generated and used alongside the [JWT Ktor authentication](jwt.md) plugin.
+[RSA (Rivest–Shamir–Adleman)](https://en.wikipedia.org/wiki/RSA_(cryptosystem)) is a widely used public-key
+crypto-system that enables secure data transmission, digital signatures, and key exchange.
 
-You can follow along by using the [auth-jwt-rs256](https://github.com/ktorio/ktor-documentation/blob/%ktor_version%/codeSnippets/snippets/auth-jwt-rs256/) sample project's private key,
-found in `resources/application.conf`. Simply save the private key in a `.pk8` file and [skip to the second](#second-step) step.
+RS256, part of the RSA encryption algorithm, utilizes SHA-256 for hashing and a key (usually
+2048-bit, 4096-bit or higher) to secure digital communications.
 
-<tip>
+In the realm of [JSON Web Token](https://jwt.io/) authentication, RS256 plays a crucial role since the integrity and
+authenticity of JWTs can be verified through signature mechanisms, such as RS256, where a public/private key pair is
+employed. This ensures that the information contained within the token remains tamper-proof and trustworthy.
+
+In this section you will learn how such keys are generated and used alongside the [Authentication JWT](jwt.md)
+plugin provided by Ktor.
+
+<warning>
 <p>
-For production use, it is recommended that you choose a more modern alternative, such as <a href="https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm">ES256</a>, that's based on more modern, efficient, and secure cryptography. However, because RSA still has its uses and is supported by the JWT plugin, the documentation below uses RSA.
+For production use, it is recommended that you choose a more modern alternative, such as <a href="https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm">ES256</a>, that's based on more efficient, and secure cryptography compared to RSA.
 </p>
-</tip>
+</warning>
 
 ## Generating an RSA private key
 
-To generate a private key, you can use `OpenSSL`, `ssh-keygen` or any other valid utility. For the rest of this page, OpenSSL will be used.
+To generate a private key, you can use OpenSSL, `ssh-keygen` or any other valid utility. For demonstration purposes,
```

Comment:
What does the word **valid** here mean?

Reply:
Perhaps better to replace "any other valid utility" with "another tool of your choice for creating authentication key pairs"?